spinalcordtoolbox/scripts/sct_smooth_spinalcord.py

```diff
@@ -210,11 +209,19 @@ def main(args=None):
 
     # Smooth the straightened image along z
     printv('\nSmooth the straightened image...')
-    sigma_smooth = ",".join([str(i) for i in sigma])
-    sct_maths.main(args=['-i', 'anat_rpi_straight.nii',
-                         '-smooth', sigma_smooth,
-                         '-o', 'anat_rpi_straight_smooth.nii',
-                         '-v', '0'])
+
+    img = Image("anat_rpi_straight.nii")
+    out = img.copy()
+
+    if len(sigmas) == 1:
+        sigmas = [sigmas[0] for i in range(len(img.data.shape))]
+    elif len(sigmas) != len(img.data.shape):
+            raise ValueError("-smooth need the same number of inputs as the number of image dimension OR only one input")
```

Comment:
I wonder if this argument validation logic would be better if it were placed earlier in `sct_smooth_spinalcord` (see next review comment)

Reply:
Sorry, this isn't actually a very useful review comment. Please disregard.